rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java

```diff
@@ -144,13 +146,160 @@ void testGetOldFederation(
 
     @Test
     void getOldFederation_previouslySetToNull_returnsNull() {
-        StorageAccessor storageAccessor = new InMemoryStorage();
-        FederationStorageProvider federationStorageProvider = new FederationStorageProviderImpl(storageAccessor);
+        FederationStorageProvider federationStorageProvider = new FederationStorageProviderImpl(null);
         federationStorageProvider.setOldFederation(null);
         Federation oldFederation = federationStorageProvider.getOldFederation(federationConstants, activations);
         assertNull(oldFederation);
     }
 
+    private static Stream<Arguments> providePendingFederationAndFormatArguments() {
+        return Stream.of(
+            Arguments.of(P2SH_ERP_FEDERATION_FORMAT_VERSION, new PendingFederationBuilder().build()),
+            Arguments.of(NON_STANDARD_ERP_FEDERATION_FORMAT_VERSION, new PendingFederationBuilder().build()),
+            Arguments.of(STANDARD_MULTISIG_FEDERATION_FORMAT_VERSION, new PendingFederationBuilder().build())
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("providePendingFederationAndFormatArguments")
+    void testGetPendingFederation(
+        int federationFormat,
+        PendingFederation expectedFederation
+    ) {
+
+        // Arrange
+
+        ActivationConfig.ForBlock activations = mock(ActivationConfig.ForBlock.class);
+
+        when(activations.isActive(ConsensusRule.RSKIP123)).thenReturn(false);
+
+        when(activations.isActive(ConsensusRule.RSKIP123)).thenReturn(true);
+
+        StorageAccessor storageAccessor = new InMemoryStorage();
+        byte[] federationFormatSerialized = getFederationFormatSerialized(federationFormat);
+        storageAccessor.saveToRepository(PENDING_FEDERATION_FORMAT_VERSION.getKey(), federationFormatSerialized);
+
+        byte[] serializedFederation = expectedFederation.serialize(activations);
+        storageAccessor.saveToRepository(PENDING_FEDERATION_KEY.getKey(), serializedFederation);
+        FederationStorageProvider federationStorageProvider = new FederationStorageProviderImpl(storageAccessor);
+
+        // Act
+
+        PendingFederation obtainedFederation = federationStorageProvider.getPendingFederation();
+
+        // Directly saving a null federation in storage to then assert that the method returns the cached federation
+        storageAccessor.saveToRepository(PENDING_FEDERATION_KEY.getKey(), null);
+
+        // Assert
+
+        // Call the method again and assert the same cached federation is returned
+        assertEquals(obtainedFederation, federationStorageProvider.getPendingFederation());
+
+        assertEquals(expectedFederation, obtainedFederation);
+
```

Comment:
i would create a separated test for this, wdyt? so every test does only one thing

Reply:
We would be duplicating exactly the same thing. This part is dependent on the above part, meaning, I would need to copy and paste the above part exactly as is anyways.